docs/feature_demultiplexer.md

```diff
@@ -0,0 +1,35 @@
+# Demultiplexer
+
+In case you need to use more pins on your microcontroller than it has available, you can use a demultiplexer. Only the 74HC138 and 74HC154 are supported.
+
+You can enable support for a demultiplexer by setting the `MATRIX_DEMUX` flag in your keyboards `rules.mk` to yes. You will also have to set the `CUSTOM_MATRIX` flag to lite.
+
+## Configuration
+
+You will need to configure which type of demultiplexer is used, which pins control its inputs and its enable pin in your keyboard's `config.h`.
+
+````
+/*
+ * Demultiplexer configuration
+ */
+
+#define DEMUX_74HC138 // This can also be DEMUX_74HC154, depending on the demultiplexer used in the keyboard.
+#define DEMUX_INPUT_PINS { D2, D1, D0 } // These are A0, A1, A2 (and A3) respectively.
+#define DEMUX_ENABLE_PIN D4
+
+/* Uncomment the line below if the enable pin needs to be high instead of low to enable the demultiplexer */
+//#define DEMUX_ENABLE_HIGH
+````
+
+You can define which column or row pins are used by the demultiplexer by just defining them in the column and row pins. Here's an example of column pins on the microcontroller and the demultiplexer:
+````
+#define MATRIX_COL_PINS { F7, F5, F6, F1, F4, F0, Y7, D5, D3, Y6, Y5, Y4, Y3, Y2, Y1, Y0 }
```

Comment:
This in particular should not be done, as the Y values can overlap with valid pin_t values.

Reply:
I realised that as I was writing the implementation, but since there are no microcontrollers with Y ports, I didn't think it would be a problem. Would a prefix like MUX_ (MUX_Y0, MUX_Y1..) be better or should it be moved to a new definition?

Reply:
No, I mean the actual `pin_t` values:
https://github.com/qmk/qmk_firmware/blob/master/quantum/config_common.h#L35-L71
in particular, on the 90USBxx/32A, port A/D starts at 0x00. We left shift by one nibble and add the pin number in the port to get a `pin_t`, so eg. `D2` would be `0x02` on the 32A.

gh60/satan for reference (D0, D1, D2, D3, D5):
![image](https://user-images.githubusercontent.com/4781841/89398375-616d5000-d754-11ea-9e16-bfa931f08eab.png)

Reply:
So if I understand correctly, Y2 would be the same as D2?

Reply:
Yes.

Reply:
Then we need to think of another address for them :)

Reply:
Well, the values go all the way up to `0xF0`, so it just isn't possible to do this.

Reply:
What do you suggest instead? I'm not very knowledgeable on the C language.

Reply:
My thought was to do something like this (I haven't tested it though):
![image](https://user-images.githubusercontent.com/4781841/89409486-80281280-d765-11ea-8291-aff25c55ba3d.png)
Then those columns would be set to `NO_PIN` in config.h:
![image](https://user-images.githubusercontent.com/4781841/89409672-c5e4db00-d765-11ea-893b-c15ad21efb7f.png)
And in the board's matrix.c, stuff like this:
![image](https://user-images.githubusercontent.com/4781841/89409602-ac439380-d765-11ea-8f10-edcee8918277.png)
Which yes, doesn't eliminate the custom matrix, but still simplifies it a decent amount.